graphql_compiler/schema_transformation/utils.py

```diff
@@ -263,3 +284,84 @@ def check_ast_schema_is_valid(ast):
 
     query_type = get_query_type_name(schema)
     visit(ast, CheckQueryTypeFieldsNameMatchVisitor(query_type))
+
+
+def is_property_field(selection):
+    """Return True if selection is a property field, False if a vertex field or type coercion.
+
+    Args:
+        selection: type Field or InlineFragment. An element occuring inside a SelectionSet
+
+    Returns:
+        True if the selection is a property field, False if it's a vertex field or type coercion
+    """
+    if isinstance(selection, ast_types.InlineFragment):
+        return False
+    if isinstance(selection, ast_types.Field):
+        if (
+            selection.selection_set is None or
+            selection.selection_set.selections is None or
+            selection.selection_set.selections == []
+        ):
+            return True
+        else:
+            return False
+    else:
+        raise AssertionError(
+            u'Input selection "{}" is not of type Field or InlineFragment.'.format(selection)
+        )
+
+
+class CheckQueryIsValidToSplitVisitor(Visitor):
```

Comment:
Would `QuerySplitValidVisitor` or some more concise version of this work? Had to read through the code to figure out what objects of this class does.